File: DavidHassen.BLL/ImageService.cs
using DavidHassen.DAL;
using DavidHassen.Shared;

namespace DavidHassen.BLL
{
    /// <summary>
    /// Service to perform Image related operations.
    /// </summary>
    public class ImageService: IImageService
    {
        #region Dependency Injection

        IImageProvider imageProvider;
        public ImageService() {
            imageProvider = new ImageProvider();
        }
        #endregion

        /// <summary>
        /// Insert new record for the uploaded image.
        /// </summary>
        /// <param name="imageModel"></param>
        /// <returns></returns>
        public bool Insert(ImageModel imageModel)
        {
            return imageProvider.Insert(imageModel);
        }

    }
}

File: Business/DavidHassen.BLL/ImageService.vb
Imports DavidHassen.DAL
Imports DavidHassen.Shared


''' <summary>
''' Service to perform Image related operations.
''' </summary>
Public Class ImageService
    Implements IImageService

#Region "Dependency Injection"

    Private imageProvider As IImageProvider

    Public Sub New()
        MyBase.New
        Me.imageProvider = New ImageProvider
    End Sub
#End Region

    Public Function Insert(ByVal imageModel As ImageModel) As Boolean Implements IImageService.Insert
        Return Me.imageProvider.Insert(imageModel)
    End Function

End Class
